spec/ruby/core/thread/thread_variables_spec.rb

```diff
@@ -13,12 +13,12 @@
     @t.thread_variable_set :a, 2
     @t.thread_variable_set :b, 4
     @t.thread_variable_set :c, 6
-    @t.thread_variables.sort.should == [:a, :b, :c]
+    @t.thread_variables.should include(:a, :b, :c)
   end
 
   it "sets a value private to self" do
-    @t.thread_variable_set :thread_variables_spec_a, 82
-    @t.thread_variable_set :thread_variables_spec_b, 82
-    Thread.current.thread_variables.should == []
+    @t.thread_variable_set :a, 82
+    @t.thread_variable_set :b, 82
+    Thread.current.thread_variables.should_not include(:a, :b)
```

Comment:
This change is fine, but then could you had a separate spec that the current Thread has no `thread_variables`?
Like
```ruby
  it "only contains user thread variables and is empty initially" do
    Thread.current.thread_variables.should == []
    @t.thread_variables.should == []
  end
```
This is useful to catch potential issues like we had here.

Reply:
For `Thread.current.thread_variables.should == []` are you relying on all other specs and code to clean any local variables they set?